specs/ReleaseChannelPreferenceSet.md

```diff
@@ -0,0 +1,325 @@
+and otherwise fall back on the stable WebView2 Runtime. With the new options you can:
+- Run automated tests against different channels on a machine with multiple channels installed
+- Target a particular channel when flighting
+- Apply different channels to different flighting rings
+- Fall back on a more stable preview channel
+
+# Examples
+## RuntimeChannelPreferenceSet
+The runtime channel preference set and channel search order are specified on the
+CoreWebView2EnvironmentOptions prior to creating the WebView2 environment.
+```c#
+CoreWebView2Environment _webViewEnvironment;
+async System.Threading.Tasks.Task CreateEnvironmentAsync() {
+    CoreWebView2RuntimeChannel channels =
+        CoreWebView2RuntimeChannel.Beta | CoreWebView2RuntimeChannel.Stable;
+    CoreWebView2EnvironmentOptions customOptions = new CoreWebView2EnvironmentOptions
+    (
+        shouldReverseChannelSearchOrder: true,
+        runtimeChannelPreferenceSet: channels
+    );
+    // Use GetAvailableCoreWebView2BrowserVersionStringWithOptions to check which
+    // channel is used with the custom options. If Beta channel was not found,
+    // install it on the device.
+    string version =
+        CoreWebView2Environment.GetAvailableBrowserVersionStringWithOptions(
+            browserExecutableFolder: null,
+            environmentOptions: customOptions);
+    if (!IsBetaChannel(version))
+    {
+        InstallChannel(CoreWebView2RuntimeChannel.Beta);
+    }
+    // If the loader is unable to find a valid installation from the runtime
+    // channel preference set, environment creation will fail.
+    _webViewEnvironment = await CoreWebView2Environment.CreateAsync(
+        options: customOptions);
+    if (_webViewEnvironment != null)
+    {
+        CheckChannel(_webviewEnvironment.BrowserVersionString());
+    }
+}
+```
+```cpp
+void AppWindow::InitializeWebViewEnvironment()
+{
+    auto options = Microsoft::WRL::Make<CoreWebView2EnvironmentOptions>();
+    CHECK_FAILURE(options->put_ShouldReverseChannelSearchOrder(TRUE));
+    COREWEBVIEW2_RUNTIME_CHANNEL channels =
+        COREWEBVIEW2_RUNTIME_CHANNEL_BETA |
+        COREWEBVIEW2_RUNTIME_CHANNEL_STABLE;
+    CHECK_FAILURE(options->SetRuntimeChannelPreferenceSet(channels));
+    // Use GetAvailableCoreWebView2BrowserVersionStringWithOptions to check which
+    // channel is used with the custom options. If Beta channel was not found,
+    // install it on the device.
+    wil::unique_cotaskmem_string version;
+    HRESULT hr = GetAvailableCoreWebView2BrowserVersionStringWithOptions(
+        nullptr, options.Get(), &version);
+    if (!IsBetaChannel(version.get()))
+    {
+        InstallChannel(COREWEBVIEW2_RUNTIME_CHANNEL_BETA);
+    }
+    // If the loader is unable to find a valid installation from the runtime
+    // channel preference set, environment creation will fail with
+    // HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND).
+    HRESULT hr = CreateCoreWebView2EnvironmentWithOptions(
```

Comment:
`hr` variable already defined.

Should we be checking the hr with CHECK_FAILURE?

Reply:
We should make sure our sample code fails with this sort of duplicate local variable (please open bug)

Please also fix sample code.